lib/literal/array.rb

```diff
@@ -106,6 +106,19 @@ def clear(...)
 		self
 	end
 
+	def concat(*value)
+		unless [Array, Literal::Array].include?(value.flatten(1).class)
+			raise ArgumentError.new("#concat expects Array arguments")
+		end
+
+		Literal.check(actual: value.flatten(1), expected: @__collection_type__) do |c|
+			c.fill_receiver(receiver: self, method: "#concat")
+		end
+
+		@__value__.concat(*value)
+		self
+	end
```

Comment:
I think we can improve this with a separate case for Literal::Array. Something like

```ruby
case
when Array
  # we need to type check here
when Literal::Array(@__type__)
  # we don’t actually need to type check because we’re guaranteed the other array has the same type as this one so we can just `@__value__.concat(other.__value__)`.
when Literal::Array
  # here, we can actually do some more checks to avoid type checking but let’s just type check for now. Could you leave a todo for me?
end
```

Reply:
Wait, `concat` takes `*value` so it’s always an Array, never a Literal::Array. 🤔

Reply:
Yeah because of the splat, `value` is always an array, and I was checking what was in _that_ array with `value.flatten(1)`

Reply:
But it’s mean to be an array of arrays so we can handle them without the type check.

Reply:
We discussed this elsewhere, but since `concat` presents some particular challenges, I've pulled it out of this PR for now 👍